xtra-libp2p/src/endpoint.rs

```diff
@@ -483,6 +488,19 @@ impl Endpoint {
                                 remote_addr,
                                 ..
                             }) => {
+                                match PeerId::try_from_multiaddr(&remote_addr) {
+                                    Some(peer_id) if blocked_peers.contains(&peer_id) => {
+                                        tracing::info!(
```

Comment:
🤔 Shouldn't we also respond with an error to the taker? I guess simply skipping here - will produce a timeout on the taker side.

Reply:
I think the intended use is to block peers that do not speak the itchysats protocol. In this case, a message wouldn't be of much use. However, maybe there are other cases?

Reply:
Isn't that scneario already handled here? 
https://github.com/itchysats/itchysats/blob/466a4efbf2cf35436fcb396be71e83f6b3fb818c/xtra-libp2p/src/endpoint.rs#L341-L343

I figured blocking peer ids would be used to block potentially malicious actors or otherwise strange behaving peers.

Reply:
That is true, but if it gets that far, does it not already entail a lot of logs that could potentially clog stuff up? Unfortunately #2527 is a bit vague on what "spamming" means. @da-kami would you be able to clarify this?

Reply:
> That is true, but if it gets that far, does it not already entail a lot of logs that could potentially clog stuff up? Unfortunately #2527 is a bit vague on what "spamming" means. @da-kami would you be able to clarify this?

At the time of writing the ticket the blacklist was meant for peers that connect but don't support *any* of our protocols. These are peers we can't do anything with - spam was meant as in log spam on the maker side, because we see those peers connecting, but then can't really perform any protocol with them. 

The idea was to be able to add `peer-id`s to a manually managed blacklist. The assumption is, that such peers actually don't run itchysats but try to connect from a different application. We obviously have to be careful not to add `peer-id`s to that list that are wrongly configured and later change their configuration to work - those would be blocked then.

That said: I don't think we can "respond" to peers when they try to establish a connection. Sending a response would require a protocol, but this is at a lower level. Unless I'm missing something we can only close the connection and this is what was planned for the ticket.

Reply:
Thanks for the clarification! In that case, can this be left as is?

Reply:
I would say 'yes'.